docs/tutorials/quantum_chess.ipynb

```diff
@@ -0,0 +1,2245 @@
+            "               │\n",
+            "(4, 5): ───X───iSwap^0.5───iSwap───────iSwap───────────────────────────────────────────────────────\n",
+            "                           │           │\n",
+            "(5, 5): ───────────────────iSwap^0.5───iSwap^0.5───iSwap───────iSwap───────────────────────────────\n",
+            "                                                   │           │\n",
+            "(6, 5): ───X───────────────────────────────────────iSwap^0.5───iSwap^0.5───────────────────────────\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "aEjS2BggQ3ke",
+        "colab_type": "text"
+      },
+      "source": [
+        "This new diagram shows how we can map the logical qubits representing 'squares' into a circuit with physical grid qubits that only uses adjacent qubits.  This circuit could then be run on a hardware device!\n",
+        "\n",
+        "**Exercise to the reader:**  Can you write the code to execute this circuit (in the simulator) and do the reverse translation back to logical named qubits to interpret the results as chess positions?"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fbyTXq9-5lLi",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Dynamic Qubit Mapping"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "vG9T-vbH5q4c",
+        "colab_type": "text"
+      },
+      "source": [
+        "The first time, we mapped the qubits from `NamedQubit`s to `GridQubit`s by hand.  Can we map qubits automatically without this hand-tuned mapping?\n",
+        "\n",
+        "We will use a depth-first search to try to automatically map the qubits.  The following code snippet is a bit longer, but it will attempt to map a qubit, followed by mapping the qubits it is connected to.  \n",
+        "\n",
+        "For example, we will first map the square 'b1' to `GridQubit` (3, 5).  We then see that 'a1', 'c1', and 'c2' need to be adjacent to 'b1', so we then map them (and their adjacent squares, etc) next.  We repeat this until we finish mapping or until we get ourselves into an impossible situation."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "0HuUkJY77wVE",
+        "colab_type": "code",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 391
+        },
+        "outputId": "6d465212-c4bb-484b-ff9b-6c00126a2e8a"
+      },
+      "source": [
+        " def map_helper(cur_node: cirq.Qid,\n",
+        "                mapping: Dict[cirq.Qid, cirq.GridQubit],\n",
+        "                available_qubits: Set[cirq.GridQubit],\n",
+        "                graph: Dict[cirq.Qid, Iterable[cirq.Qid]]) -> bool:\n",
+        "        \"\"\"Helper function to construct mapping.\n",
```

Comment:
Optional nit: Per google python style guide, add one more new-line to leave a blank line between the first line of the docstring and the rest of it; same applies before Args: and before Returns: below.